docs/migrating-from-0.9.md

```diff
@@ -0,0 +1,251 @@
+
+Refer to https://js.api.genkit.dev/modules/_genkit-ai_express.html for more details.
+
+`startFlowServer` has moved from part of the genkit object to the express package, to use startFlowServer, update your imports. 
+
+
+**Old:**
+
+```ts
+const ai = genkit({ ... });
+ai.startFlowServer({
+  flows: [myFlow1, myFlow2],
+});
+```
+
+**New**
+
+```ts
+import { startFlowServer } from '@genkit-ai/express';
+startFlowServer({
+  flows: [myFlow1, myFlow2],
+});
+```
+
+## Changes to Flows
+
+`run` function for custom trace blocks has moving to `ai.run`
+
+**Old:**
+
+```ts
+ai.defineFlow({name: 'banana'}, async (input, streamingCallback) => {
+  const step = await run('myCode', async () => {
+    return 'something'
+  });
+})
+```ts
+
+**New**
+
+```ts
+ai.defineFlow({name: 'banana'}, async (input, {context, sendChunk}) => {
+  const step = await ai.run('myCode', async () => {
+    return 'something'
+  });
+})
+```
+
+`ai.defineStreamingFlow` has been removed. Use `ai.defineFlow` instead. Also, `streamingCallback` moved to a field inside the second argument of he flow function and is now called `sendChunk`.
+
+**Old:**
+
+```ts
+const flow = ai.defineStreamingFlow({name: 'banana'}, async (input, streamingCallback) => {
+  streamingCallback({chunk: 1});
+})
+
+const {stream} = await flow()
+for await (const chunk of stream) {
+  // ...
+}
+```
+
+**New**
+
+```ts
+const flow = ai.defineFlow({name: 'banana'}, async (input, {context, sendChunk}) => {
+  sendChunk({chunk: 1});
+})
+
+const {stream, output} = flow.stream(input);
+for await (const chunk of stream) {
+  // ...
+}
+```
+
+Flow auth is now called context. You can access auth as a field inside context:
+
+**Old:**
+
+```ts
+ai.defineFlow({name: 'banana'}, async (input) => {
+  const auth = getFlowAuth();
+  // ...
+})
+```
+
+**New**
+
+```ts
+ai.defineFlow({name: 'banana'}, async (input, { context }) => {
+  const auth = context.auth;
+})
+```
+
+`onFlow` moved to `firebase-functions/https` package and got renamed to `onCallGenkit`. Here's how to use it:
+
+```ts
+import { onCallGenkit } from "firebase-functions/https";
+import { defineSecret } from "firebase-functions/params";
+import { genkit, z } from "genkit";
+
+const apiKey = defineSecret("GOOGLE_GENAI_API_KEY");
+
+const ai = genkit({
+  plugins: [googleAI()],
+  model: gemini15Flash,
+});
+
+export const jokeTeller = ai.defineFlow(
+  {
+    name: "jokeTeller",
+    inputSchema: z.string().nullable(),
+    outputSchema: z.string(),
+    streamSchema: z.string(),
+  },
+  async (type, { sendChunk }) => {
+    const { stream, response } = ai.generateStream(
+      `Tell me a longish ${type ?? "dad"} joke.`
+    );
+    for await (const chunk of stream) {
+      sendChunk(chunk.text);
+    }
+    return (await response).text;
+  }
+);
+
+export const tellJoke = onCallGenkit({ secrets: [apiKey] }, jokeTeller);
+```
+
+## Prompts
+
+We've make several improvements to prompts.
```

Comment:
These are very cool features, but are the backward incompatible? If not, I'd suggest moving to a seperate "new features" section to make it clear which section of the doc is "you have to do this" vs "you might want to do this"

Reply:
+1, I think it would help to break the guide into two sections:
- Namespace/syntax changes to keep existing code working
- New features since 0.9 (genkit express server, context, prompt enhancements)

Currently these are interwoven throughout.

Reply:
the prompts api is a breaking change....